### PR TITLE
refresh round_date() docs

### LINKED ISSUE #850: Refresh `round_date()` docs
* [ ] Proofread title, description, and parameters 
* [ ] Proofread comments in examples 
* [ ] Ensure comment headings used to divide into scannable sections 
* [ ] Flag examples that seem overly complex for further review 
* [ ] Switch out mtcars/iris for something more interesting, or a smaller dataset constructed specifically to illustrate one point. 
* [ ] Review size of example outputs to ensure that you can easily see important rows and cols. 
* [ ] Datasets should always be tibbles to avoid use of `as_tibble()` distracting from the main point

### PR DESCRIPTION
**Pull request by sushmitavgopalan16**
fixes #850 

- there are no examples printing out datasets 
- mtcars isn't used at all

### CODE REVIEW

File: R/round.r [hunk 1]
```diff
@@ -50,17 +52,33 @@
...
-#'   Date Objects` below for more details.
-#' @param week_start when unit is `weeks` specify the reference day; 7 being Sunday.
+#' @param change_on_boundary if this is `NULL` (the default), instants on the boundary
+#'   remain unchanged, but `Date` objects are rounded up to the next boundary.
+#'   If this is `TRUE`, instants on the boundary are rounded up to the next boundary.
+#'   If this is `FALSE`, nothing on the boundary is rounded up at all. This was the
+#'   default for \pkg{lubridate} prior to `v1.6.0`.
+#'   See section `Rounding Up Date Objects` below for more details.
+#'
+#'   For example, as per default behaviour `NULL`,
```
Comment (krlmlr) [inline R/round.r]: I like the way the behavior is detailed here. Could you please convert to examples and add code just before the `@export` ?

File: R/round.r [hunk 2]
```diff
@@ -117,6 +119,21 @@
 #' as.POSIXct("2009-08-03 12:01:59.3") ## -> "2009-08-03 12:01:59.2 CEST"
 #' ceiling_date(x, ".1 sec") ## -> "2009-08-03 12:01:59.2 CEST"
 #'
+#' ## behaviour of change_on_boundary
```
Comment (krlmlr) [inline R/round.r]:
```suggestion
#' ## behaviour of `change_on_boundary`
```

**Comment by krlmlr** [comment]
@hadley: Can you please merge?

**Comment by hadley** [comment]
Thanks @sushmitavgopalan16!

**Pull request merged**